### DISCUSSION TITLE
Monorepo version constraints with SCM versioning

### DISCUSSION THREAD

**Discussion by elventear** [pdm-project/pdm#2533] [Q&A]
Hi,

I am using PDM to develop in a monorepo that will publish multiple python build products. I am also using SCM versioning in the monorepo. I am trying to figure out how can I express a dependency between packages, such that the SCM version that is used for the build product can also be used as a constraint between dependencies.

For example, if I have packages `A` and `B` on the monorepo that depend on each other , i.e.`A -> B`, and the SCM version is `0.1.dev8+gb3c95c6.d20240102`, I would like `B` to express as a dependency `A==0.1.dev8+gb3c95c6.d20240102`, when the package is built. And ideally this is all managed dynamically. Is there a way to express this sort of relationship?

**Comment by stoh563** [comment]
I have just come across the same problem, is there a solution? If required, I will be happy to provide more information.

**Comment by elventear** [comment]
I haven't solved the problem due to lack of time, but I have thought that building some PDM plugin that changes the dependency would be the solution. As source of inspiration I was looking at https://github.com/pdm-project/pdm-build-locked.
